main.go

```diff
@@ -94,3 +107,43 @@ func download(filepath string, url string) error {
 	_, err = io.Copy(out, resp.Body)
 	return err
 }
+
+func extractFiles(filePaths []string) ([]string, error) {
+	var filenames []string
+	for i, filename := range filePaths {
+		r, err := zip.OpenReader(filename)
+		if err != nil {
+			return nil, fmt.Errorf("error while extracting zip files")
+		}
+		defer r.Close()

```

Comment:
Esse difer só vai rolar depois que sair da função (ele não rola a cada iteração do for loop)

se quiser fazer isso você deve criar uma função e chamar aqui, ou criar uma função anônima.